specification/hardwaresecuritymodules/HardwareSecurityModules.Management/models.tsp

```diff
@@ -0,0 +1,851 @@
+
+  /**
+   * The Cloud HSM Cluster public network access
+   */
+  publicNetworkAccess?: PublicNetworkAccess;
+
+  /**
+   * Cloud HSM Cluster status message
+   */
+  @visibility(Lifecycle.Read)
+  statusMessage?: string;
+}
+
+/**
+ * The Cloud HSM Properties
+ */
+model CloudHsmProperties {
+  /**
+   * FQDN of the Cloud HSM
+   */
+  fqdn?: string;
+
+  /**
+   * The Cloud HSM State. Values are: Deploying, ConfiguringSlb, Starting, Starting, Failed, Failed, Deleting, DeletingSlbEntry, InitialProvisioning, Updating
+   */
+  state?: string;
+
+  /**
+   * The Cloud HSM State message
+   */
+  stateMessage?: string;
+}
+
+/**
+ * Properties of the private endpoint connection.
+ */
+model PrivateEndpointConnectionProperties {
+  /**
+   * The private endpoint resource.
+   */
+  privateEndpoint?: CommonTypes.PrivateEndpoint;
+
+  /**
+   * A collection of information about the state of the connection between service consumer and provider.
+   */
+  privateLinkServiceConnectionState: CommonTypes.PrivateLinkServiceConnectionState;
+
+  /**
+   * The provisioning state of the private endpoint connection resource.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: PrivateEndpointConnectionProvisioningState;
+
+  /**
+   * The group ids for the private endpoint resource.
+   */
+  @visibility(Lifecycle.Read)
+  groupIds?: string[];
+}
+
+/**
+ * The private endpoint resource.
+ */
+model PrivateEndpoint {
+  /**
+   * The ARM identifier for private endpoint.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+}
+
+/**
+ * A collection of information about the state of the connection between service consumer and provider.
+ */
+model PrivateLinkServiceConnectionState {
+  /**
+   * Indicates whether the connection has been Approved/Rejected/Removed by the owner of the service.
+   */
+  status?: PrivateEndpointServiceConnectionStatus;
+
+  /**
+   * The reason for approval/rejection of the connection.
+   */
+  description?: string;
+
+  /**
+   * A message indicating if changes on the service provider require any updates on the consumer.
+   */
+  actionsRequired?: string;
+}
+
+/**
+ * Common fields that are returned in the response for all Azure Resource Manager resources
+ */
+model Resource {
+  /**
+   * Fully qualified resource ID for the resource. E.g. "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}"
+   */
+  @visibility(Lifecycle.Read)
+  id?: Azure.Core.armResourceIdentifier;
+
+  /**
+   * The name of the resource
+   */
+  @visibility(Lifecycle.Read)
+  name?: string;
+
+  /**
+   * The type of the resource. E.g. "Microsoft.Compute/virtualMachines" or "Microsoft.Storage/storageAccounts"
+   */
+  @visibility(Lifecycle.Read)
+  type?: string;
+
+  /**
+   * Azure Resource Manager metadata containing createdBy and modifiedBy information.
+   */
+  @visibility(Lifecycle.Read)
+  systemData?: SystemData;
+}
+
```

Comment:
@MattGertz @rkmanda @raosuhas

Redeclaring such a custom 'Resource' type in typespec (or anything in common types, but Resource most obviously) really should need a *justification* or suppression. It shouldn't just be allowed by default.

Which raises the question to me, why don't we get a linter error for this? Seems like an obvious gap!